pychromecast/socket_client.py

```diff
@@ -1133,7 +1142,14 @@ def new_socket() -> socket.socket:
     Try to set SO_REUSEPORT for BSD-flavored systems if it's an option.
     Catches errors if not.
     """
-    new_sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        new_sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+        # ensuring dual-stack
+        new_sock.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, 0)
+    except OSError:
+        # falling back to IPv4 on systems without IPv6
+        new_sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
     new_sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
```

Comment:
This looks strange to me. We are not making sure the socket type matches the data returned from getaddrinfo. The creation of the socket should be postponed until we have resolved the name using getaddrinfo

Reply:
There isn't really a need to match the socket to the resolved address type, as the dual-stack socket can connect to both types. If it falls back to a v4-only socket and only a v6 address is returned, it would fail to connect either way,